src/main/java/jenkins/branch/BuildRetentionBranchProperty.java

```diff
@@ -84,6 +85,7 @@ public Boolean run() {
     /**
      * Our {@link hudson.model.Descriptor}.
      */
+    @Symbol("buildRetention")
```

Comment:
The `DataBoundConstructor` for this class includes `BuildDiscarder` as an argument. `BuildDiscarder` is an abstract class, so I was not sure if I needed to add `Symbol` annotation for `BuildDiscarder` subclasses?  The same question applies to some other classes below.

Reply:
Yes, ideally we would track down all implementations of `BuildDiscarder` and make sure they have symbols as well, but the only implementation I am aware of is `LogRotator` and that already has a symbol, see [here](https://github.com/jenkinsci/jenkins/blob/ab5616712ab8d79bcabc9a97e10a0edb2f5696c4/core/src/main/java/hudson/tasks/LogRotator.java#L304).

That said, I don't think that `BuildRetentionBranchProperty` is intended to be directly configured by users because it doesn't show up in the UI when configuring a property strategy for a multibranch projects (not sure why, I don't see an `isApplicable` override in the descriptor), so maybe we shouldn't add a symbol for it?

Reply:
I do see a jelly file for `BuildRetentionBranchProperty` [here](https://github.com/jenkinsci/branch-api-plugin/tree/master/src/main/resources/jenkins/branch/BuildRetentionBranchProperty)
Or does  having a jelly file not always preclude a  UI?

Reply:
@car-roll Yeah, I think they might show up in the UI in some context, I'm just not sure what those contexts are, because I don't see them as options when I configure a property strategy for a Multibranch Pipeline. 

`UntrustedBranchProperty` implements [`BranchPropertyDescriptor.isApplicable`](https://github.com/jenkinsci/branch-api-plugin/blob/b99766caece8edb67adc0ce2dde8e3f0d2df67c5/src/main/java/jenkins/branch/UntrustedBranchProperty.java#L143-L152) so that it only shows up when the items inside of the `MultiBranchProject` extend `hudson.model.Project`, but `WorkflowJob` directly extends `hudson.model.Job`, so it makes sense that that one didn't show up in my testing.

After some digging it looks like Multibranch Pipelines have a [`DescriptorVisibilityFilter`](https://github.com/jenkinsci/workflow-multibranch-plugin/blob/97747eb6361453df2d2d181f55b24243d2f7fa7e/src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java#L223-L233) that suppresses `BuildRetentionBranchProperty` and `RateLimitBranchProperty`, so that explains why I didn't see them either in my testing.

So as far as I can tell, adding symbols to these types doesn't matter for Multibranch Pipelines, but we could still go ahead and add them for whatever projects do support them.

Reply:
`RateLimitBranchProperty` already has a symbol  annotation  so we're good to go there.

Reply:
Looks like the symbol is only for [the descriptor for the job property](https://github.com/jenkinsci/branch-api-plugin/blob/b99766caece8edb67adc0ce2dde8e3f0d2df67c5/src/main/java/jenkins/branch/RateLimitBranchProperty.java#L393-L395), but I think we want one on [the descriptor for the branch property](https://github.com/jenkinsci/branch-api-plugin/blob/b99766caece8edb67adc0ce2dde8e3f0d2df67c5/src/main/java/jenkins/branch/RateLimitBranchProperty.java#L186).